nevergrad/functions/causaldiscovery/core.py

```diff
@@ -0,0 +1,89 @@
+import typing as tp
+import networkx as nx
+import cdt
+from cdt.metrics import (precision_recall, SHD)
+import nevergrad as ng
+from ..base import ExperimentFunction
+
+
+# pylint: disable=too-many-instance-attributes,too-many-arguments,too-many-statements,too-many-locals
+class CausalDiscovery(ExperimentFunction):
+    """
+    All parameters, except for the `generator` argument, are used to configure `acylicgraph`.
+    For the setting of the acylic graph generator, please refer to https://fentechsolutions.github.io/CausalDiscoveryToolbox/html/data.html
+
+    Parameters
+    ----------
+    generator: str
+        'tuebingen', 'sachs', 'dream4' or 'acylicgraph'
+    causal_mechanism: str
+        'linear', 'polynomial', 'sigmoid_add', 'sigmoid_mix', 'gp_add', 'gp_mix', 'nn'
+    noise: str or function
+        Type of noise function to use in the generative process
```

Comment:
FYI allowing functions will not allow recording it as a descriptor, which are used for the benchmarks. It is not strictly a problem, but it may get different experiments mixed up without it